src/chanakya/core/memory_management.py

```
@@ -30,7 +41,9 @@ def retrieve_relevant_memories(user_message, limit=3):
     if not keywords:
         conn.close()
         return []
-    query = "SELECT datetime, memory FROM memories WHERE " + " OR ".join([f"memory LIKE ?" for _ in keywords])
+    query = "SELECT datetime, memory FROM memories WHERE " + " OR ".join(
+        ["memory LIKE ?" for _ in keywords]
+    )
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)